docs/user/filter-and-process/README.md

```diff
@@ -1,16 +1,31 @@
 # Filtering and Processing Data
 
-When you configure the inputs for your logs, traces, and metrics, you can choose from which specific Kubernetes resources you want to include or exclude data. Your data is automatically transformed and enriched so you can analyze it in your OTLP backend.
+The Telemetry module provides several ways to refine your telemetry data. You can control what data is collected, enrich it with valuable context, modify its structure, and drop irrelevant information before it reaches your backend.
 
-## Filtering Mechanisms
+This processing happens at two distinct stages in the data lifecycle:
 
-The Telemetry module supports the following mechanisms to filter the data, which apply at different stages of the collection process:
+## Input-Level Filtering
 
-- Filtering within a pipeline (LogPipeline, MetricPipeline): You can configure the `input` section of a pipeline to select or reject data before it is processed by the agent. This is the most common method for filtering application logs, runtime metrics, and Prometheus metrics.
-- Configuring the data source (Istio `Telemetry` CRD): For Istio-generated data (access logs and traces), you configure the Istio `Telemetry` resource itself. This controls which workloads generate data and at what volume (sampling rate), before it is even sent to a pipeline.
+You can select or reject data at the source, before it enters a pipeline. This is useful for controlling data volume from specific namespaces, applications, or Kubernetes resources. This method is efficient as it prevents unwanted data from being processed at all.
 
-## Processing Data
+The Telemetry module supports the following input-level filtering mechanisms:
 
-Application logs from containers are automatically transformed into structured OpenTelemetry (OTLP) log records.
+- **Pipeline input filtering** (LogPipeline, MetricPipeline): Configure the `input` section of a pipeline to select or reject data before it is processed by the agent. This is the most common method for filtering application logs, runtime metrics, and Prometheus metrics.
+- **Source-level filtering** (Istio `Telemetry` CRD): For Istio-generated data (access logs and traces), configure the Istio `Telemetry` resource itself to control which workloads generate data and at what volume (sampling rate).
 
-All pipelines automatically enrich telemetry data with Kubernetes resource attributes, such as Pod name, namespace, and labels. With this context information, you can easily identify the source of telemetry data in your backend.
+For details, see [Filter Logs](filter-logs.md), [Filter Metrics](filter-metrics.md), and [Filter Traces](filter-traces.md).
```

Comment:
The details section should be moved into the list. And links for the second case will be usefull as well.